test_with_postgres.sh

```
@@ -0,0 +1,229 @@
+
+# Step 3: Run database migrations
+echo -e "\n${YELLOW}Step 3: Running database migrations${NC}"
+uv run alembic upgrade head
+echo -e "${GREEN}✓ Migrations complete${NC}"
+
+# Step 4: Run Python-based critical fixes test
+echo -e "\n${YELLOW}Step 4: Running critical fixes verification${NC}"
+chmod +x test_critical_fixes.py
+uv run python test_critical_fixes.py
+
+if [ $? -eq 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

(SC2181)